src/UPSTransitRequest.php

```diff
@@ -0,0 +1,191 @@
+<?php
+
+namespace Drupal\commerce_ups;
+
+use const COMMERCE_UPS_LOGGER_CHANNEL;
+use DateInterval;
+use DateTime;
+use Drupal;
+use Drupal\commerce_shipping\Entity\ShipmentInterface;
+use Drupal\Core\Logger\LoggerChannelFactoryInterface;
+use Drupal\physical\WeightUnit;
+use Ups\Entity\AddressArtifactFormat;
+use Ups\Entity\InvoiceLineTotal;
+use Ups\Entity\Shipment;
+use Ups\Entity\ShipmentWeight;
+use Ups\Entity\TimeInTransitRequest;
+use Ups\Entity\UnitOfMeasurement;
+use Ups\TimeInTransit;
+
+/**
+ * Class to fetch the transit time for a shipment.
+ *
+ * @package Drupal\commerce_ups
+ */
+class UPSTransitRequest extends UPSRateRequest implements UPSTransitRequestInterface {
+
+  /**
+   * The configuration array from a CommerceShippingMethod.
+   *
+   * @var array
+   */
+  protected $configuration;
```

Comment:
**CHANGE REQUEST**

No need to redefine properties already defined in the parent i.e. `UPSRateRequest`. Nevertheless, something seems off here; it extends a parent class that receives the child as an argument in its constructor, was it meant to extend `UPSRequest` instead of `UPSRateRequest`?

Reply:
Yeah, that makes sense. I think that was a typo because we don't need anything from RateRequest.